src/utils/periods/relativePeriods.js

```diff
@@ -195,3 +240,118 @@ export const flattenedRelativePeriods = RELATIVE_PERIODS.reduce(
 
 export const isFixedPeriodType = periodType =>
     !!fixedPeriodTranslations[periodType]
+
+const formatYyyyMmDd = timestamp => {
+    const date = new Date(timestamp)
+    const yyyy = date.getFullYear()
+    const mm = (date.getMonth() + 1).toString().padStart(2, '0')
+    const dd = date
+        .getDate()
+        .toString()
+        .padStart(2, '0')
+
+    return `${yyyy}-${mm}-${dd}`
+}
+
+const getCurrentDate = () => new Date(Date.now())
+
+const getPreviousMonday = date => {
+    const day = date.getDay()
+    const mondayTimeStamp = date.getDate() - day + (day === 0 ? -6 : 1)
```

Comment:
This is a bit hard to parse.

Reply:
I agree it's hard to digest, but I don't really see a way around this. "Our" weeks start on a Monday but `date.getDay()` will return `0` for Sunday, so a "real/normal" week starts on Sunday. My conclusion is that this is hard to parse, because it's a bit quirky/complex.

Reply:
Yeah makes sense. We could quote your explanation in a comment. That would already help me grasp more quickly what's going on.

Reply:
Added in eb454bb